mlflow/models/signature.py

```diff
@@ -383,19 +403,25 @@ def _infer_signature_from_type_hints(
     except InvalidTypeHintException as e:
         warnings.warn(e.message, stacklevel=2)
         return None
-    is_output_type_hint_valid = type_hints.output is not None
-    try:
-        output_schema = (
-            _infer_schema_from_type_hint(type_hints.output) if type_hints.output else None
-        )
-    except InvalidTypeHintException as e:
-        warnings.warn(
-            f"Invalid output type hint, setting output schema to AnyType. Error: {e}", stacklevel=2
-        )
-        is_output_type_hint_valid = False
-        output_schema = Schema([ColSpec(type=AnyType())])
+
+    default_output_schema = Schema([ColSpec(type=AnyType())])
+    is_output_type_hint_valid = False
+    output_schema = None
+    if type_hints.output:
+        try:
+            output_schema = _infer_schema_from_type_hint(type_hints.output)
+            is_output_type_hint_valid = True
+        except InvalidTypeHintException as e:
+            warnings.warn(
+                f"Invalid output type hint, setting output schema to AnyType. Error: {e}",
```

Comment:
"Invalid" sounds too assertive (nothing wrong with the type hint itself). Can we say "Unsupported"? Also I would downgrade this to INFO level, because most of the case setting AnyType does no harm and users don't need to fix it.